Lib/tempfile.py

```diff
@@ -270,6 +270,20 @@ def _mkstemp_inner(dir, pre, suf, flags, output_type):
     raise FileExistsError(_errno.EEXIST,
                           "No usable temporary file name found")
 
+def _dont_follow_symlinks(func, path, *args):
+    # Pass follow_symlinks=False, unless not supported on this platform.
+    if func in _os.supports_follow_symlinks:
+        func(path, *args, follow_symlinks=False)
+    elif _os.name == 'nt' or not _os.path.islink(path):
+        func(path, *args)
+
+def _resetperms(path):
+    try:
+        _dont_follow_symlinks(_os.chflags, path, 0)
+    except AttributeError:
+        pass
```

Comment:
The scope of the try/except AttributeError is unclear to me. I would prefer to write the code like that, since it now calls the non-trivial _dont_follow_symlinks() function:

```suggestion
    try:
        chflags = _os.chflags
    except AttributeError:
        pass
    else:
        _dont_follow_symlinks(chflags, path, 0)
```